airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -228,12 +228,16 @@ public SourceDiscoverSchemaRead discoverSchemaForSourceFromSourceId(final Source
   public SourceDiscoverSchemaRead discoverSchemaForSourceFromSourceCreate(final SourceCoreConfig sourceCreate)
       throws ConfigNotFoundException, IOException, JsonValidationException {
     final StandardSourceDefinition sourceDef = configRepository.getStandardSourceDefinition(sourceCreate.getSourceDefinitionId());
+    final var partialConfig = configRepository.statefulSplitEphemeralSecrets(
+        sourceCreate.getConnectionConfiguration(),
+        sourceDef.getSpec());
+
     final String imageName = DockerUtils.getTaggedImageName(sourceDef.getDockerRepository(), sourceDef.getDockerImageTag());
     // todo (cgardens) - narrow the struct passed to the client. we are not setting fields that are
     // technically declared as required.
     final SourceConnection source = new SourceConnection()
         .withSourceDefinitionId(sourceCreate.getSourceDefinitionId())
-        .withConfiguration(sourceCreate.getConnectionConfiguration());
```

Comment:
Is there way that we could have programmatically verified that we weren't passing any secrets in here, which would have prevented us from writing this method this way in the first place? Something like having dedicated types for configs that contain secrets vs those that don't?

Just trying to brainstorm how we could avoid making this same mistake again in the future

Reply:
yeah. i think the suggestion you made is the direction we should go. we can even in the openapi spec, set the configuration field to a class called `SecretsContainingConfiguration` that just wraps the json object to make it super obvious.